microcluster/app.go

```diff
@@ -34,7 +34,8 @@ import (
 type MicroCluster struct {
 	FileSystem *sys.OS
 
-	args Args
+	args   Args
+	daemon *daemon.Daemon
```

Comment:
There looks the be only one place where we access the daemon directly and that is during the call to `Start()` as this setups the local microcluster deamon.

All the other client funcs (`Ready()`, `SQL()`, `JoinCluster()` to name a few) use API calls to retrieve the information.
I am almost certain we have to go the same route with the list of API extensions.

Inside the API handler you can access the daemon via the its `State()`. See [here](https://github.com/canonical/microcluster/blob/main/internal/daemon/daemon.go#L808) how to pass something from the daemon which otherwise cannot be reached.

Maybe it would make sense to add the list to the `/cluster/1.0` endpoint like we have in LXD when querying `/1.0`. What do you think @masnax ?
You can then query this endoint in the `APIExtensions()` func and return the respective value.

Reply:
Actually, there is no need for doing that... The extension list is already available when calling `GetClusterMembers` . I just forgot it was exposed there :)

Reply:
@roosterfish updated :)

Reply:
Great thanks, looks much cleaner :)

Reply:
fixed